### ISSUE TITLE
kaggle tpu kernel(tf 2.2.0) "Socket closed" when using subclassing api 

### ISSUE THREAD

**Issue by chenghuige** [tensorflow/tensorflow#38538]
When doing model.fit, will always got socket lost/closed error using tpu kernel V3-8 if using subclassing api while functional api is ok. code like below

    class Model(keras.Model):
        def __init__(self):
            super(Model, self).__init__() 

            bert_layer = hub.load(BERT_GCS_PATH_SAVEDMODEL)
            bert_layer = hub.KerasLayer(bert_layer, trainable=True)
            self.bert_layer = bert_layer
            self.dense = keras.layers.Dense(1)

       def call(self, input):
            input_word_ids = input['input_word_ids']
            input_mask = input['input_mask']
            segment_ids = input['all_segment_id']
  
            x, _ = self.bert_layer([input_word_ids, input_mask, segment_ids])
            x = self.dense(x)
            self.logit = x
            x = tf.math.sigmoid(x)
            return x

**Comment by Saduf2019** [comment]
@chenghuige 
i ran the code shared by you on nightly and do not face any issues, please let me know if i am missing anything here, [gist for the same](https://colab.sandbox.google.com/gist/Saduf2019/34d4a5419f146e6fb08665f8eb587643/38538.ipynb)
please share the error faced by you.

**Comment by chenghuige** [comment]
@Saduf2019  I tried to use lates tf in kaggle kernel, following [https://www.kaggle.com/kivlichangoogle/jigsaw-multilingual-getting-started](jigsaw-multilingual-getting-started) but with lates tf, tpu setup will fail, I have modified the colab file you provide.

**Comment by gowthamkpr** [comment]
@chenghuige I would recommend you to post this in [Kaggle/kaggle-api](https://github.com/Kaggle/kaggle-api/issues) as this platform is only for tensorflow issues. Thanks!

**Comment by gowthamkpr** [comment]
Closing this issue as its not related to tensorflow. Please post it in the kaggle platform. Thanks!
